src/client/components/MealWithId.js

```diff
@@ -0,0 +1,125 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import { useParams } from "react-router";
+import AddReservation from "./AddReservation";
+import AddReview from "./AddReview";
+import LoadImage from "../LoadImage";
+import Displaymeal from "./Displaymeal";
+import { useHistory } from 'react-router'
+
+const MealWithId = () => {
+  const [mealWithId, setMealWithId] = useState([]);
+  const [error, setError] = useState(null);
+  const [loading, setLoading] = useState(true);
+  const [reservationForm, setReservationForm] = useState(false);
+  const [reviewForm, setReviewForm] = useState(false);
+  const [reservation, setReservation] = useState([]);
+  const paramMealId = useParams();
+  const history = useHistory()
+  useEffect(() => {
+    fetch(`/api/meals?availableReservations=true`)
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else {
+          throw new Error("Something went wrong!");
+        }
+      })
+      .then((data) => {
+        setReservation(data);
+        setLoading(false);
+      })
+      .catch((error) => {
+        setError(error.message);
+        setLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    fetch(`/api/meals/${paramMealId.id}`)
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        } else {
+          throw new Error("Something went wrong!");
+        }
+      })
+      .then((data) => {
+        setMealWithId(data[0]);
+        setLoading(false);
+      })
+      .catch((error) => {
+        setError(error.message);
+        setLoading(false);
+      });
+  }, []);
+
+  let getReservationWithId = reservation.filter(
```

Comment:
Instead of filter which returns an array, you could use find which return the first element that matches the condition.
That way you don't have to do later on getReservationWithId**[0]**